tephra-hbase-compat-1.1-base/src/main/java/org/apache/tephra/hbase/txprune/PruneUpperBoundWriter.java

```diff
@@ -79,15 +72,17 @@ public void run() {
         while (!isInterrupted()) {
           long now = System.currentTimeMillis();
           if (now > (lastChecked + pruneFlushInterval)) {
-            if (shouldFlush.compareAndSet(true, false)) {
+            if (!pruneEntries.isEmpty()) {
               // should flush data
               try {
-                dataJanitorState.savePruneUpperBoundForRegion(regionName, pruneUpperBound.get());
+                Iterator<PruneInfo> iterator = pruneEntries.iterator();
```

Comment:
Since we are doing concurrent operations here it would be good to use `peek()` and `poll()`, rather than a iterator.